config/routes.rb

```diff
@@ -0,0 +1,17 @@
+Rails.application.routes.draw do
+  root 'welcome#index'
+  devise_for :users
+  
+  get 'welcome', to: 'welcome#index', as: :welcome
+  patch 'recipes/:id/update_public', to: 'recipes#update_public', as: 'recipe_update_public'
+  
+  resources :recipes, only: %i[index show new create destroy] do
+    patch :update_public, on: :member
+  end
+
+  resources :recipes
+  resources :public_recipes, only: [:index]
+  resources :recipe_foods, only: %i[create new destroy], path: 'recipe_foods'
+  resources :foods, only: [:index, :new, :create, :destroy]
+  get 'general_shopping_list', to: 'shopping_list#index', as: :shopping_list
+end
```

Comment:
- [ ] We are highly encouraged not to use any unnecessary routes in our app. To clean up the unused routes, you can remove the redundant lines from the routes.rb file and remain with the following:

```
Rails.application.routes.draw do
  root 'welcome#index'
  devise_for :users

  patch 'recipes/:id/update_public', to: 'recipes#update_public', as: 'recipe_update_public'

  resources :recipes, only: %i[index show new create destroy] do
    patch :update_public, on: :member
  end

  resources :public_recipes, only: [:index]
  resources :recipe_foods, only: %i[create new destroy]
  resources :foods, only: [:index, :new, :create, :destroy]
  get 'general_shopping_list', to: 'shopping_list#index', as: :shopping_list
end
```